src/main/scala/loamstream/loam/intake/aggregator/Metadata.scala

```diff
@@ -1,35 +1,200 @@
-    varIdFormat: Option[String],
+    varIdFormat: String = Defaults.varIdFormat,
     ancestry: String,
-    author: Option[String],
+    author: Option[String] = None,
     tech: String,
-    cases: Int,
-    controls: Int) {
+    quantitative: Metadata.Quantitative) {
     
-  varIdFormat.foreach(f => require(VarIdFormat.isValid(f)))
+  require(VarIdFormat.isValid(varIdFormat))
   
-  def subjects: Int = cases + controls
+  def subjects: Int = quantitative.subjects
   
   def asConfigFileContents: String = {
     val authorPart = author.map(a => s"author ${a}").getOrElse("")
     
-    val varIdFormatPart = varIdFormat.map(f => s"var_id ${f}").getOrElse("")
+    import Metadata.Quantitative.CasesAndControls
+    
+    val casesPart = quantitative match {
+      case CasesAndControls(cases, _) => s"cases ${cases}"
+      case _ => ""
+    }
+    
+    val controlsPart = quantitative match {
+      case CasesAndControls(_, controls) => s"controls ${controls}"
+      case _ => ""
+    }
     
     s"""|dataset ${dataset} ${phenotype}
         |ancestry ${ancestry}
         |tech ${tech}
-        |cases ${cases}
-        |controls ${controls}
+        |${casesPart}
+        |${controlsPart}
         |subjects ${subjects}
-        |${varIdFormatPart}
-        |${authorPart}""".stripMargin
+        |var_id ${varIdFormat}
+        |${authorPart}""".stripMargin.trim
+  }
+}
+
+object Metadata extends ConfigParser[Metadata] {
+  sealed trait Quantitative {
+    def subjects: Int
+  }
+  
+  object Quantitative {
+    final case class CasesAndControls(cases: Int, controls: Int) extends Quantitative {
+      override def subjects: Int = cases + controls
+    }
+    
+    final case class Subjects(value: Int) extends Quantitative {
+      override def subjects: Int = value
+    }
+  }
+  
+  private final case class Parsed(
+    dataset: String,
+    phenotype: Option[String] = None,
+    varIdFormat: String = Defaults.varIdFormat,
+    ancestry: String,
+    author: Option[String] = None,
+    tech: String,
+    cases: Option[Int] = None,
+    controls: Option[Int] = None,
+    subjects: Option[Int] = None) {
+    
+    private def quantitative: Try[Quantitative] = (cases, controls, subjects) match {
+      case (Some(cas), Some(cntrls), None) => Success(Quantitative.CasesAndControls(cas, cntrls))
+      case (None, None, Some(subjects)) => Success(Quantitative.Subjects(subjects))
+      case _ => Tries.failure(s"Either both cases and controls OR subjects must be supplied")
+    }
+    
+    def toMetadata: Try[Metadata] = {
+      for {
+        ph <- Options.toTry(phenotype)("Expected phenotype to be present")
+        q <- quantitative
+      } yield {
+        Metadata(
+          dataset = dataset,
+          phenotype = ph,
+          varIdFormat = varIdFormat,
+          ancestry = ancestry,
+          author = author,
+          tech = tech,
+          quantitative = q)
+      }
+    }
+    
+    def toNoPhenotype: Try[NoPhenotype] = {
```

Comment:
What's the `NoPhenotype` possibility here? I ask since that would be an error when running the intake code (a phenotype is required).

Reply:
I'm glad you noticed this.  Some of this config-parsing code is still messy, left over from when I was trying to get data-uploads working.  The issue here is that I was given a bunch of files and corresponding phenotypes.  I wanted a way to specify in a config file the metadata that was common between all those uploads.